key-vault/azure-key-vault/src/main/java/com/quorum/tessera/key/vault/AzureKeyVaultClientFactory.java

```diff
@@ -0,0 +1,35 @@
+package com.quorum.tessera.key.vault;
+
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.rest.credentials.ServiceClientCredentials;
+
+import javax.annotation.PreDestroy;
+import java.util.concurrent.ExecutorService;
+
+
+/**
+ * Authenticates to Azure Key Vault by providing a callback to authenticate
+ * using adal.
+ */
+public class AzureKeyVaultClientFactory {
+
+    private final ServiceClientCredentials serviceClientCredentials;
+
+    private ExecutorService executorService;
+
+    public AzureKeyVaultClientFactory(ExecutorService executorService) {
+        String clientId = System.getenv("AZURE_CLIENT_ID");
```

Comment:
Can we avoid System.getenv() assuming this object will be wired in can we access the environment vars in the spring config.